File: VB/DVCustomExportOption/XtraReport1.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports System.Collections.Generic
Imports DevExpress.XtraPrinting
Imports System.Drawing.Imaging
Imports System.IO

Namespace DVCustomExportOption
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport

		Public Sub New()
			InitializeComponent()
			InitializeComponent()
			Dim persons As New List(Of Person)()
			persons.Add(New Person() With {.PersonFirstName = "Max", .PersonLastName = "Fowler", .PersonAge = 30})
			persons.Add(New Person() With {.PersonFirstName = "Nancy", .PersonLastName = "Drewmore", .PersonAge = 29})
			persons.Add(New Person() With {.PersonFirstName = "Pak", .PersonLastName = "Jang", .PersonAge = 31})

			Me.DataSource = persons
			Me.CreateDocument()
		End Sub
	End Class

	Public Class Person
		Public Property PersonFirstName() As String
		Public Property PersonLastName() As String
		Public Property PersonAge() As Integer
	End Class
End Namespace

File: CS/DVCustomExportOption/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using System.Collections.Generic;
using DevExpress.XtraPrinting;
using System.Drawing.Imaging;
using System.IO;

namespace DVCustomExportOption {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
            InitializeComponent();
            List<Person> persons = new List<Person>();
            persons.Add(new Person() { PersonFirstName = "Max", PersonLastName = "Fowler", PersonAge = 30 });
            persons.Add(new Person() { PersonFirstName = "Nancy", PersonLastName = "Drewmore", PersonAge = 29 });
            persons.Add(new Person() { PersonFirstName = "Pak", PersonLastName = "Jang", PersonAge = 31 });

            this.DataSource = persons;
            this.CreateDocument();
        }
    }

    public class Person {
        public string PersonFirstName { get; set; }
        public string PersonLastName { get; set; }
        public int PersonAge { get; set; }
    }
}
